### PR TITLE
add fix to postgresql provider for bbox across 180°

### PR DESCRIPTION
**Pull request by benoitdm-oslandia**
When the feature request has a bbox filter across the 180°, it failed to select the features. When the `QgsCoordinateTransform::transformBoundingBox` function is called the `handle180Crossover` parameter was not set to `true`.

Also add a check in `QgsPostgresFeatureIterator::whereClauseRect` to handle correct bbox coordinates when xMin > xMax

[X] should be backported

### CODE REVIEW

File: src/providers/postgres/qgspostgresfeatureiterator.cpp [hunk 1]
```diff
@@ -474,6 +474,13 @@ bool QgsPostgresFeatureIterator::close()
 QString QgsPostgresFeatureIterator::whereClauseRect()
 {
   QgsRectangle rect = mFilterRect;
+
+  // in case coordinates are around 180.0°
+  if ( rect.xMinimum() > rect.xMaximum() )
```
Comment (nyalldawson) [inline src/providers/postgres/qgspostgresfeatureiterator.cpp:479]: I think this should also be testing that the geometries ARE geographic, or it'll potentially cause issues later (if we add support for non-geographic wrap arounds of the date line)

Comment (benoitdm-oslandia) [reply src/providers/postgres/qgspostgresfeatureiterator.cpp:479]: Sure! But what is the best? `mSource->mCrs.isGeographic()` or `mSource->mSpatialColType == SctGeography`

**Comment by benoitdm-oslandia** [comment]
reopen please

**Comment by benoitdm-oslandia** [comment]
reopen please

**Pull request closed without merging**